# Ebay Used Car Sales Project

The aim of this project is to clean the data and analyze the used car listings. We have a dataset that was originally uploaded to Kaggle but adjustments was made to closely resemble a scraped dataset. EBay is a shopping site where people can buy and sell a wide variety of goods and services. To learn more about the company, EBay, click [here](https://en.wikipedia.org/wiki/EBay)




In [36]:
import pandas as pd
import numpy as np
autos = pd.read_csv('autos.csv',encoding= 'Latin-1')

In [37]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

There a total of 20 columns. Dataset contains two data types stored as strings or integers. The column names are in camelcases.

# Cleaning the columns

We'll start by cleaning the column names to make the data easier to work with. First, we will change the column names from camelcase to snakecase.

In [38]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [39]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen']
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price  ab_test vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  num_photos  postal_code  \
0     peugeot              nein  2016-03-26 00:00:00           0        79588   
1         bmw              nein  2016-04-04 00:00:00           0        71034   
2  volkswagen              nein  2016-03-26 00:00:00           0        35394   
3       smart              nein  2016-03-12 00:00:00           0        33729   
4        ford              nein  2016-04-01 00:00:00           0        39218   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

# Initial data exploration

We will use the describe() function to generate descriptive statistics for all columns of the input will be included in the output.

In [40]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price ab_test  \
count                 50000        50000   50000      50000  50000   50000   
unique                48213        38754       2          2   2357       2   
top     2016-03-30 17:37:35  Ford_Fiesta  privat    Angebot     $0    test   
freq                      3           78   49999      49999   1421   25756   
mean                    NaN          NaN     NaN        NaN    NaN     NaN   
std                     NaN          NaN     NaN        NaN    NaN     NaN   
min                     NaN          NaN     NaN        NaN    NaN     NaN   
25%                     NaN          NaN     NaN        NaN    NaN     NaN   
50%                     NaN          NaN     NaN        NaN    NaN     NaN   
75%                     NaN          NaN     NaN        NaN    NaN     NaN   
max                     NaN          NaN     NaN        NaN    NaN     NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  num_photos   postal_code            last_seen  
count                 50000     50000.0  50000.000000                50000  
unique                   76         NaN           NaN                39481  
top     2016-04-03 00:00:00         NaN           NaN  2016-04-07 06:17:27  
freq                   1946         NaN           NaN                    8  
mean                    NaN         0.0  50813.627300                  NaN  
std                     NaN         0.0  25779.747957                  NaN  
min                     NaN         0.0   1067.000000                  NaN  
25%                     NaN         0.0  30451.000000                  NaN  
50%                     NaN         0.0  49577.000000                  NaN  
75%                     NaN         0.0  71540.000000                  NaN  
max                     NaN         0.0  99998.000000                  NaN

Initial observation:
- num_photos column does not make sense
- seller and offer_type columns have the same values

In [41]:
autos["num_photos"].value_counts()

0    50000
Name: num_photos, dtype: int64

Based on the .value_counts() the num_photos column has 0 for every column. We'll drop this column, plus the other two (seller and offer_type columns) we noted above have the same values. So, let's go ahead and drop these 3 columns as shown below.


In [42]:
autos = autos.drop(["num_photos", "seller", "offer_type"], axis=1)
autos.columns

Index(['date_crawled', 'name', 'price', 'ab_test', 'vehicle_type',
       'registration_year', 'gearbox', 'power_ps', 'model', 'odometer',
       'registration_month', 'fuel_type', 'brand', 'unrepaired_damage',
       'ad_created', 'postal_code', 'last_seen'],
      dtype='object')

Next, we see that price column and odometer column are numeric values stored as text. We will remove non-numeric characters and convert both columns into numeric data type

In [43]:
autos['price'] = autos["price"].str.replace("$","").str.replace(",","").astype(int)
autos['price'].head()

0    5000
1    8500
2    8990
3    4350
4    1350
Name: price, dtype: int64

In [44]:
autos['odometer'] = autos['odometer'].str.replace(',',"").str.replace("km","").astype(int)
autos['odometer'].head()

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer, dtype: int64

In [45]:
autos.rename({"odometer":"odometer_km"}, axis=1, inplace=True)

In [46]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 17 columns):
date_crawled          50000 non-null object
name                  50000 non-null object
price                 50000 non-null int64
ab_test               50000 non-null object
vehicle_type          44905 non-null object
registration_year     50000 non-null int64
gearbox               47320 non-null object
power_ps              50000 non-null int64
model                 47242 non-null object
odometer_km           50000 non-null int64
registration_month    50000 non-null int64
fuel_type             45518 non-null object
brand                 50000 non-null object
unrepaired_damage     40171 non-null object
ad_created            50000 non-null object
postal_code           50000 non-null int64
last_seen             50000 non-null object
dtypes: int64(6), object(11)
memory usage: 6.5+ MB


#  Exploring Odometer and Price

After cleaning up the two columns (Odometer and Price), lets dig deeper into exploring the data!

In [47]:
autos["odometer_km"].head()

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer_km, dtype: int64

In [48]:
autos["odometer_km"].unique().shape

(13,)

In [49]:
autos["odometer_km"].value_counts()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64

From the observation above, the values are rounded up. This might indicate that sellers had to choose from pre-set options for this field. Additionally, there are more high mileage than low mileage vehicles available at Ebay. Next, we will look at the price field.

In [50]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [51]:
autos["price"].unique().shape

(2357,)

In [52]:
autos["price"].value_counts().head(10)

0       1421
500      781
1500     734
2500     643
1000     639
1200     639
600      531
800      498
3500     498
2000     460
Name: price, dtype: int64

The prices in this field are rounded up as well. There are 2357 unique value listings. There are 1421 cars listed with $0 price. The maximum price is one hundred million dollars, which seems a lot, let's look at the highest prices further. Lets sort the values in ascending and descending order to see how customers list the cars.

In [53]:
autos["price"].value_counts().sort_index(ascending=False).head(20)


99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: price, dtype: int64

In [54]:
autos["price"].value_counts().sort_index(ascending=True).head(20)

0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
13       2
14       1
15       2
17       3
18       1
20       4
25       5
29       1
30       7
35       1
Name: price, dtype: int64

Given that eBay is an auction site, there could legitimately be items where the opening bid starts from $1. We will keep the 1 dollar items, and remove anything above 350,000 dollars, since it seems that prices increase steadily to that number and then jump up to less realistic numbers, which are 999990 dollars and above.

In [55]:
autos = autos[autos["price"].between(1,351000)]
autos["price"].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price, dtype: float64

The above cell gives us a maximum price point of 350000 dollars and minimum price point of 1 dollar. Goal achieved!

# Exploring the date columns
There are a number of columns with date information:
- date_crawled
- registration_month
- registration_year
- ad_created (exclude this)
- last_seen

We'll explore each of these columns to learn more about the listings.

We will look at these columns identified as string values and we will convert the data into numerical values to understand it quantitatively. These three columns are identified as string values as follows date_crawled , last_seen, ad_created

In [56]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48565 entries, 0 to 49999
Data columns (total 17 columns):
date_crawled          48565 non-null object
name                  48565 non-null object
price                 48565 non-null int64
ab_test               48565 non-null object
vehicle_type          43979 non-null object
registration_year     48565 non-null int64
gearbox               46222 non-null object
power_ps              48565 non-null int64
model                 46107 non-null object
odometer_km           48565 non-null int64
registration_month    48565 non-null int64
fuel_type             44535 non-null object
brand                 48565 non-null object
unrepaired_damage     39464 non-null object
ad_created            48565 non-null object
postal_code           48565 non-null int64
last_seen             48565 non-null object
dtypes: int64(6), object(11)
memory usage: 6.7+ MB


In [57]:
autos[['date_crawled','ad_created','last_seen']]

date_crawled           ad_created            last_seen
0      2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1      2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2      2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3      2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4      2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50
5      2016-03-21 13:47:45  2016-03-21 00:00:00  2016-04-06 09:45:21
6      2016-03-20 17:55:21  2016-03-20 00:00:00  2016-03-23 02:48:59
7      2016-03-16 18:55:19  2016-03-16 00:00:00  2016-04-07 03:17:32
8      2016-03-22 16:51:34  2016-03-22 00:00:00  2016-03-26 18:18:10
9      2016-03-16 13:47:02  2016-03-16 00:00:00  2016-04-06 10:46:35
10     2016-03-15 01:41:36  2016-03-14 00:00:00  2016-04-07 03:16:21
11     2016-03-16 18:45:34  2016-03-16 00:00:00  2016-03-16 18:45:34
12     2016-03-31 19:48:22  2016-03-31 00:00:00  2016-04-06 14:17:52
13     2016-03-23 10:48:32  2016-03-23 00:00:00  2016-04-01 14:17:13
14     2016-03-23 11:50:46  2016-03-23 00:00:00  2016-04-01 15:46:47
15     2016-04-01 12:06:20  2016-04-01 00:00:00  2016-04-02 21:10:48
16     2016-03-16 14:59:02  2016-03-16 00:00:00  2016-03-18 05:29:37
17     2016-03-29 11:46:22  2016-03-29 00:00:00  2016-04-05 20:46:26
18     2016-03-26 19:57:44  2016-03-26 00:00:00  2016-04-02 12:16:41
19     2016-03-17 13:36:21  2016-03-17 00:00:00  2016-03-17 14:45:58
20     2016-03-05 19:57:31  2016-03-05 00:00:00  2016-03-07 05:46:46
21     2016-03-06 19:07:10  2016-03-06 00:00:00  2016-04-05 23:46:19
22     2016-03-28 20:50:54  2016-03-28 00:00:00  2016-04-01 06:45:30
23     2016-03-10 19:55:34  2016-03-10 00:00:00  2016-03-17 08:45:17
24     2016-04-03 11:57:02  2016-04-03 00:00:00  2016-04-07 13:16:50
25     2016-03-21 21:56:18  2016-03-21 00:00:00  2016-04-01 05:16:49
26     2016-04-03 22:46:28  2016-04-03 00:00:00  2016-04-05 23:46:48
28     2016-03-19 21:56:19  2016-03-19 00:00:00  2016-04-07 14:58:48
29     2016-04-02 12:45:44  2016-04-02 00:00:00  2016-04-02 12:45:44
30     2016-03-14 11:47:31  2016-03-14 00:00:00  2016-03-14 11:47:31
...                    ...                  ...                  ...
49968  2016-04-01 17:49:15  2016-04-01 00:00:00  2016-04-05 13:16:35
49969  2016-03-17 18:49:02  2016-03-17 00:00:00  2016-03-25 23:18:15
49970  2016-03-21 22:47:37  2016-03-21 00:00:00  2016-04-07 04:17:34
49971  2016-03-29 14:54:12  2016-03-29 00:00:00  2016-03-29 20:41:51
49972  2016-03-26 22:25:23  2016-03-26 00:00:00  2016-03-28 11:28:18
49973  2016-03-27 05:32:39  2016-03-27 00:00:00  2016-03-27 08:25:24
49975  2016-03-27 20:51:39  2016-03-27 00:00:00  2016-04-05 19:45:34
49976  2016-03-19 18:56:05  2016-03-19 00:00:00  2016-04-07 06:16:44
49977  2016-03-31 18:37:18  2016-03-31 00:00:00  2016-04-06 12:16:11
49978  2016-04-04 10:37:14  2016-04-04 00:00:00  2016-04-06 12:44:20
49979  2016-03-20 18:38:40  2016-03-20 00:00:00  2016-04-07 01:45:12
49980  2016-03-12 10:55:54  2016-03-12 00:00:00  2016-04-06 17:16:49
49981  2016-03-15 09:38:21  2016-03-15 00:00:00  2016-04-05 17:21:46
49982  2016-03-29 18:51:08  2016-03-29 00:00:00  2016-03-29 18:51:08
49983  2016-03-06 12:43:04  2016-03-06 00:00:00  2016-03-09 06:16:08
49985  2016-04-02 16:38:23  2016-04-02 00:00:00  2016-04-06 15:17:52
49986  2016-04-04 20:46:02  2016-04-04 00:00:00  2016-04-06 23:16:00
49987  2016-03-22 20:47:27  2016-03-22 00:00:00  2016-03-26 22:46:06
49988  2016-03-28 19:49:51  2016-03-28 00:00:00  2016-04-07 02:17:40
49989  2016-03-11 19:50:37  2016-03-11 00:00:00  2016-03-12 10:17:55
49990  2016-03-21 19:54:19  2016-03-21 00:00:00  2016-04-06 22:46:57
49991  2016-03-06 15:25:19  2016-03-06 00:00:00  2016-03-06 18:24:19
49992  2016-03-10 19:37:38  2016-03-10 00:00:00  2016-03-13 01:44:51
49993  2016-03-15 18:47:35  2016-03-15 00:00:00  2016-04-06 19:46:53
49994  2016-03-22 17:36:42  2016-03-22 00:00:00  2016-04-06 08:16:39
49995  2016-03-27 14:38:19  2016-03-27 00:00:00  2016-04-01 13:47:40
49996  2016-

Noticed that these columns have date and time stamp. We only want the date range in order to understand the data, we can extract the date values.

In [58]:
autos['date_crawled'].str[:10].value_counts(normalize=True, 
                                dropna=False).sort_index()
# normalize set to True, returns the relative 
# frequency by dividing all values by the sum of values
# dropna set to False, shows NaN index values as well
# .sort_index(), rank date in ascending order (earliest to latest)

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

In [59]:
autos['date_crawled'].str[:10].value_counts(normalize=True, 
                                dropna=False).sort_values()

2016-04-07    0.001400
2016-04-06    0.003171
2016-03-18    0.012911
2016-04-05    0.013096
2016-03-06    0.014043
2016-03-13    0.015670
2016-03-05    0.025327
2016-03-24    0.029342
2016-03-16    0.029610
2016-03-27    0.031092
2016-03-25    0.031607
2016-03-17    0.031628
2016-03-31    0.031834
2016-03-10    0.032184
2016-03-26    0.032204
2016-03-23    0.032225
2016-03-11    0.032575
2016-03-22    0.032987
2016-03-09    0.033090
2016-03-08    0.033296
2016-03-30    0.033687
2016-04-01    0.033687
2016-03-29    0.034099
2016-03-15    0.034284
2016-03-19    0.034778
2016-03-28    0.034860
2016-04-02    0.035478
2016-03-07    0.036014
2016-04-04    0.036487
2016-03-14    0.036549
2016-03-12    0.036920
2016-03-21    0.037373
2016-03-20    0.037887
2016-04-03    0.038608
Name: date_crawled, dtype: float64

The site was crawled daily over roughly a one month period in March and April 2016. The distribution of listings crawled on each day is roughly uniform.

Maybe last_seen column could give us an idea on when the listing was sold or removed from Ebay.

In [60]:
autos['last_seen'].str[:10].value_counts(normalize=True, 
                                dropna=False).sort_index()

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

The last three days starting from 2016-04-05 to 2016-04-07 
contain a disproportionate amount of 'last seen' values. It is unlikely that there was a massive spike in sales, and more likely that these values are to do with the crawling period ending and don't indicate car sales.

In [61]:
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

The year that the car was first registered will likely indicate the age of the car. Looking at this column, we note some odd values. The minimum value is 1000 and the maximum is 9999, these values are not realistic for the registration year. Lets organize the year of registration data in more details.

# Fixing the vehicles' year of registration
As stated in the above cell, vehicles with registration year above 2016 is not correct because it would not make sense for the car listings to be on Ebay. As such, we would decide on acceptable values for the vehicles' year of registration.

We can try isolating the registration_year column into a new variable, being a series. Then in that series you can safely drop any outliers. We can get the statistics based on the new variable excluding outliers.

In [75]:
autos['registration_year'].isnull().sum()
# there is zero null values in the column

0

In [78]:
valid_yrs = autos['registration_year']
valid_yrs = valid_yrs[valid_yrs.between(1900, 2016)]

In [89]:
valid_yrs.value_counts().sort_index(ascending=True).head(10)


1910    5
1927    1
1929    1
1931    1
1934    2
1937    4
1938    1
1939    1
1941    2
1943    1
Name: registration_year, dtype: int64

In [88]:
valid_yrs.value_counts().sort_index(ascending=False).head(10)



2016    1220
2015     392
2014     663
2013     803
2012    1310
2011    1623
2010    1589
2009    2085
2008    2215
2007    2277
Name: registration_year, dtype: int64

In [90]:
reg_year_only.describe()


count    46681.000000
mean      2002.910756
std          7.185103
min       1910.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       2016.000000
Name: registration_year, dtype: float64

We can see that most of the vehicles are registered 2000s onwards. 2008 is the year of registration with the highest number of vehicles registered.